students/ShujinCao/lesson09/assignment/charges_calc.py

```diff
@@ -0,0 +1,129 @@
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+import logging
+
+
+def main():
+    """
+    main function
+    """
+    log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+    log_file = datetime.datetime.now().strftime("%Y-%m-%d") + ".log"
+    args = parse_cmd_arguments()
+    if args.debug == 1:
+        logging.basicConfig(level=logging.ERROR, format=log_format,
+                            filename=log_file, filemode="a+")
+    if args.debug == 2:
+        logging.basicConfig(level=logging.WARNING, format=log_format,
+                            filename=log_file, filemode="a+")
+    if args.debug == 3:
+        logging.basicConfig(level=logging.DEBUG, format=log_format,
+                            filename=log_file, filemode="a+")
+    input_data = load_rentals_file(args.input)
+    logging.warning("DATA READ IN DONE - CALCULATION START")
+    output_data = calculate_additional_fields(input_data)
+    logging.warning("CALCULATION DONE - SAVE OUTPUT TO JSON FILE")
+    save_to_json(args.output, output_data)
+    logging.warning("OUTPUT HAS BEEN SUCCESFULLY SAVED")
+
+def parse_cmd_arguments():
+    """
+    add arguments
+    """
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument(
+        '-o',
+        '--output',
+        help='ouput JSON file',
+        required=True)
+    parser.add_argument(
+        "-d",
+        "--debug",
+        type=int,
+        choices=range(
+            0,
+            4),
+        help="0: No debug messages or log file.\
+                         1: Only error messages.\
+                         2: Error messages and warnings.\
+                         3: Error messages, warnings and debug messages")
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    load json file
+    """
+    with open(filename) as file:
+        try:
+            json_f = json.load(file)
+        except BaseException:
+            logging.error("DATA WAS NOT LOADED.")
+            exit(0)
+    return json_f
+
+def turn_off_logging(func):
+
+    def decorated_func(*args, **kwargs):
+        logging.disable(logging.CRITICAL)
+        print("some logging has been disabled")
+        returned_value = func(*args, **kwargs)
+        logging.disable(logging.NOTSET)
+        print("some logging has been re-enabled")
+        return returned_value
+
+    return decorated_func
```

Comment:
Decorator looks good. Doesn't toggle from the argparse arguments, but that was not clear in the assignment, tricky to do, and seldom necessary.